### PR TITLE
Resolves #84- Migrate UI to TS

### LINKED ISSUE #84: Migrate frontend to typescript
The frontend is in JS right now, it would be nice to migrate it to typescript.

### PR DESCRIPTION
**Pull request by BeNikk**
Converted all frontend files to TS

### CODE REVIEW

File: frontend/src/components/pages/DecryptPassword/DecryptPassword.tsx [hunk 1]
```diff
@@ -69,11 +87,11 @@ const DecryptPassword = () => {
   };
 
   useEffect(() => {
-    if (decryptedData.length > 0) {
+    if (typeof decryptedData == "string" && decryptedData.length > 0) {
```
Comment (frankhart2018) [inline frontend/src/components/pages/DecryptPassword/DecryptPassword.tsx:90]: Why do you need to check the type here?

File: frontend/src/components/pages/DeviceInfo/DeviceInfo.tsx [hunk 2]
```diff
@@ -9,21 +9,37 @@ import {
...
-  const dispatch = useDispatch();
+  const dispatch = useDispatch<any>();
 
-  const convertCToF = (tempVal) => {
+  const convertCToF = (tempVal: number) => {
     return ((9 * tempVal) / 5 + 32).toFixed(1);
   };
 
-  const formatTemp = (tempVal) => {
+  const formatTemp = (tempVal: any) => {
```
Comment (frankhart2018) [inline frontend/src/components/pages/DeviceInfo/DeviceInfo.tsx:42]: Why is this any, this is either a string or a number.

File: frontend/src/services/password-thunk.ts [hunk 3]
```diff
@@ -1,10 +1,41 @@
 import { createAsyncThunk } from "@reduxjs/toolkit";
 
 import * as passwordService from "./password-service";
+interface PasswordPayload {
```
Comment (frankhart2018) [inline frontend/src/services/password-thunk.ts:4]: These should probably go to a separate models directory.

File: frontend/package.json [hunk 4]
```diff
@@ -43,6 +43,14 @@
     ]
   },
```
Comment (frankhart2018) [inline frontend/package.json:44]: Please fix all the react warnings.

**Review by frankhart2018 — COMMENTED** [review]
This looks awesome. Some minor changes, nothing too concerning. You can make those changes if you want, but I'm fine with merging as is too.

**Comment by BeNikk** [comment]
Sure sir, I'll have a look at it in evening

**Pull request closed without merging**